src/multidim/setdiff.jl

```diff
@@ -44,65 +29,21 @@ Algorithm: Start from the total overlap (in all directions);
 expand each direction in turn.
 """
 function setdiff(A::IntervalBox{N,T}, B::IntervalBox{N,T}) where {N,T}
-    X = [labelled_setdiff(a, b) for (a, b) in zip(A.v, B.v)]
-    # ordered such that the first in each is the excluded interval
-
-    first = [ i[1] for i in X ]
-    labels = [i[2] for i in first]
-
-    any(labels .== -1) && return [A]  # no overlap
-
-    # @assert all(labels .== 1)
-
-    excluded = [i[1] for i in first]
 
-    result_list = IntervalBox{N,T}[]
-
-    for dimension in N:-1:1
-        for which in X[dimension][2:end]
-            excluded[dimension] = which[1]
-            push!(result_list,
-                IntervalBox(excluded[1:dimension]..., A[dimension+1:N]...))
+    intersection = A ∩ B
+    isempty(intersection) && return [A]
+
+    result_list = fill(IntervalBox(emptyinterval(T), N), 2*N)
```

Comment:
You could instead try to make an empty array, use `sizehint`, then `push!` to it.

Reply:
this seemed to produce performance regression (wrt the improved version after your previous comment). Particularly it introduces one extra allocation and the time goes to 104ns. Here's how I implemented it
```
    result_list = sizehint!(IntervalBox{N, T}[], 2*N)
    x = A.v
    @inbounds for i = 1:N
        tmp = _setdiff(A[i], B[i])
        @inbounds for j = 1:2
            if !isempty(tmp[j])
                x = setindex(x, tmp[j], i)
                push!(result_list, IntervalBox{N, T}(x))
            end
        end
        x = setindex(x, intersection[i], i)
    end
    return result_list
    ```